demo/addons/fileimport/fileimport.js

```diff
@@ -0,0 +1,499 @@
+        })
+    })];
+
+    var _resetForms = function () {
+        $("#geocoding-modal .csv-fields a").remove();
+        $("#x-select option").remove();
+        $("#y-select option").remove();
+        $("#srs-select option").remove();
+    }
+
+    var _initCoordsTab = function (tmp, oLayer) {
+        // init xy select options
+        var options = [];
+        tmp.meta.fields.forEach(function (f) {
+            options.push(`<option value="${f}">${f}</option>`);
+        });
+        $("#x-select").append(options.join(" "));
+        $("#y-select").append(options.join(" "));
+
+        // take last two fields as coords by default
+        $("#x-select").val(tmp.meta.fields[tmp.meta.fields.length - 2]);
+        $("#y-select").val(tmp.meta.fields[tmp.meta.fields.length - 1]);
+
+        // init and update xy data preview
+        $("#x-data").text(tmp.data[0][$("#x-select").val()]);
+        $("#y-data").text(tmp.data[0][$("#y-select").val()]);
+
+        $("#x-select").change(function (e) {
+            $("#x-data").text(tmp.data[0][e.target.value]);
+        })
+        $("#y-select").change(function (e) {
+            $("#y-data").text(tmp.data[0][e.target.value]);
+        })
+
+        // switch select option values and data preview via button
+        $("#btn-xy").off().on("click", function () {
+            var xToY = $("#x-select").val();
+            $("#x-select").val($("#y-select").val()).trigger("change");
+            $("#y-select").val(xToY).trigger("change");
+        });
+
+        // init srs select option
+        var srs = ["<option value='EPSG:4326'>EPSG:4326</option>"];
+        if (oLayer.projections && oLayer.projections.projection) {
+            oLayer.projections.projection.forEach(function (p) {
+                //register projections
+                proj4.defs(p.proj4js);
+                ol.proj.proj4.register(proj4);
+                //display projections
+                var epsg = p.proj4js.split(",")[0].replace(/['"]+/g, '');
+                srs.push(`<option value="${epsg}">${epsg}</option>`);
+            });
+        }
+        $("#srs-select").append(srs.join(" "));
+    }
+
+    // POST local file and parameters to API geocode service
+    var _geocode = function (_csv, oLayer, l) {
+        if (oLayer.geocoder === "ban") {
+            // Create post form data
+            var formData = new FormData();
+            formData.append("data", new Blob([_csv], { "type": "text/csv" }));
+            if (!oLayer.geocodingcitycode) {
+                oLayer.geocodingfields.forEach(function (value) {
+                    formData.append("columns", value);
+                });
+            } else {
+                formData.append("citycode", oLayer.geocodingcitycode);
+            }
+            $("#csv-status").attr("class", "wait");
+            $.ajax({
+                type: "POST",
+                processData: false,
+                url: "https://api-adresse.data.gouv.fr/search/csv/",
```

Comment:
Need to appear in config file as options.

Reply:
I think it's already present in param `geocoder` which includes more than the `url` (handling the response)

Reply:
So you prefer to keep hard URL code  ?

Just imagine base API URL change. Admin user (no dev profile) couldn't fix that.

I prefer to centralize all URL in config.

Reply:
> Taken the fact that the import only supports one geocoder

It's true.

> we could perhaps use the geocoder param

Is it possible to use geocoder without xml config ? If not, use geocoder param. Else use addon config.

But you could by default use addon URL and read config.xml param if set.... just an idea.

Reply:
Having had a look at it again, I added a param `geocoderurl' to allow an admin to override the url incase the ban url changes or he wants to use an API that behaves like the ban API. I kept it optional to minimize the mandatory params. In parallel I conserved the `geocoder` param to allow the implementation of a different API "type" in the future...